TELCODOCS-1707: Add preparing/configuring shared varlibcontainers (#77256)
openshift/openshift-docs: modules/cnf-image-based-upgrade-share-container-directory.adoc

// Module included in the following assemblies:
// * edge_computing/image-based-upgrade/cnf-preparing-for-image-based-upgrade.adoc

:_mod-docs-content-type: PROCEDURE
[id="cnf-image-based-upgrade-shared-container-directory_{context}"]

> = Configuring a shared container directory between `ostree` stateroots

Reviewer: Remove backticks in headings (per [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata)):
```suggestion
= Configuring a shared container directory between ostree stateroots
```

Apply a `MachineConfig` to both the seed and the target clusters during installation time to create a separate partition and share the `/var/lib/containers` directory between the two `ostree` stateroots that will be used during the upgrade process.

[IMPORTANT]
====
You must complete this procedure at installation time.
====

.Procedure

> * Apply a `MachineConfig` to create a separate partition.

Reviewer: "Use a colon at the end of the lead-in sentence" (per ISG):
```suggestion
* Apply a `MachineConfig` to create a separate partition:
```

      disks:
        - device: /dev/disk/by-id/wwn-<root_disk> <1>
          partitions:
            - label: varlibcontainers
              startMiB: <start_of_partition> <2>
              sizeMiB: <partition_size> <3>
      filesystems:
        - device: /dev/disk/by-partlabel/varlibcontainers
          format: xfs
          mountOptions:
            - defaults
            - prjquota
          path: /var/lib/containers
          wipeFilesystem: true
    systemd:
      units:
        - contents: |-
            # Generated by Butane
            [Unit]
            Before=local-fs.target
            Requires=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service
            After=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service

            [Mount]
            Where=/var/lib/containers
            What=/dev/disk/by-partlabel/varlibcontainers
            Type=xfs
            Options=defaults,prjquota

            [Install]
            RequiredBy=local-fs.target
          enabled: true
          name: var-lib-containers.mount
----
<1> Specify the root disk.
<2> Specify the start of the partition in MiB. If the value is too small, the installation will fail.

> <3> Specify a minimum size for the partition of 500GB to ensure adequate disk space for precached images. If the value is too small, the deployments after installation will fail.

Reviewer: Per ISG and [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#gb):
```suggestion
<3> Specify a minimum size for the partition of 500 GB to ensure adequate disk space for precached images. If the value is too small, the deployments after installation will fail.
```
